apps/site/lib/site/shuttle_diversion.ex

```diff
@@ -214,4 +210,37 @@ defmodule Site.ShuttleDiversion do
   defp stops_with_direction(%{attributes: %{"direction_id" => direction_id}} = trip) do
     trip |> stops() |> Enum.map(&{direction_id, &1})
   end
+
+  defp trips_on_affected_routes(trips) do
+    Stream.filter(trips, &shuttle_related?/1)
+  end
+
+  defp trips_on_unaffected_routes(route_ids, trips, time) do
+    # Since trips can only be retrieved by service date and not by a finer-grained time window,
+    # the overall collection of trips may include ones that are both typical and atypical for the
+    # routes experiencing a diversion, if the diversion occurs only during part of the day. So we
+    # need to reject trips that are typical, but belong to a route that has an active diversion.
+
+    affected_route_ids =
+      route_ids
```

Comment:
Do we need to compute `affected_route_ids` this way as opposed to using the trips from `trips_on_affected_routes/1` and getting the route ids from there?

Reply:
I think it'd end up being the same set of IDs; there's no cost to doing it one way or the other, but I do like using the alert data to make decisions where possible, since they're the most reliable indicator we have that something is actually a shuttle service (when we have things not "reachable" from an alert, like the shapes, we have to make an educated guess).